src/hooks/UseRepository.ts

```diff
@@ -0,0 +1,13 @@
+import {useQuery} from '@tanstack/react-query';
+import {fetchRepositoryDetails} from 'src/resources/RepositoryResource';
+
+export function useRepository(org: string, repo: string) {
+  const {data, error} = useQuery(['repodetails', org, repo], () =>
+    fetchRepositoryDetails(org, repo),
+  );
+
+  return {
```

Comment:
We should export the loading variable from here as well. we can also put initial for the request here which helps prevent null errors elsewhere

Reply:
Added